testworld/signature_test.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHost_ValidSignature(t *testing.T) {
+	// Hosts
+	bob := doctorFord.getHostTestSuite(t, "Bob")
+	eve := doctorFord.getHostTestSuite(t, "Eve")
+
+	ctxh := testingconfig.CreateAccountContext(t, eve.host.config)
+
+	// Get PublicKey and PrivateKey
+	publicKey, privateKey := GetSigningKeyPair(t, eve.host.idService, eve.id, ctxh)
+
+	collaborators := [][]byte{bob.id[:]}
+	dm := createCDWithEmbeddedPO(t, collaborators, eve.id, publicKey, privateKey)
+
+	signatures, signatureErrors, err := eve.host.p2pClient.GetSignaturesForDocument(ctxh, dm)
+	assert.NoError(t, err)
+	assert.Nil(t, signatureErrors)
+	assert.Equal(t, 1, len(signatures))
+}
+
+func TestHost_FakedSignature(t *testing.T) {
+	// Hosts
+	alice := doctorFord.getHostTestSuite(t, "Alice")
+	bob := doctorFord.getHostTestSuite(t, "Bob")
+	eve := doctorFord.getHostTestSuite(t, "Eve")
+
+	actxh := testingconfig.CreateAccountContext(t, alice.host.config)
+	ectxh := testingconfig.CreateAccountContext(t, eve.host.config)
+
+	// Get PublicKey and PrivateKey
+	publicKey, privateKey := GetSigningKeyPair(t, alice.host.idService, alice.id, actxh)
+
+	collaborators := [][]byte{bob.id[:]}
+	dm := createCDWithEmbeddedPO(t, collaborators, eve.id, publicKey, privateKey)
+
+	signatures, signatureErrors, err := eve.host.p2pClient.GetSignaturesForDocument(ectxh, dm)
+	assert.NoError(t, err)
+	assert.Error(t, signatureErrors[0], "Signature verification failed error")
+	assert.Equal(t, 0, len(signatures))
+}
+
+func TestHost_RevokedSigningKey(t *testing.T) {
+	// Hosts
+	bob := doctorFord.getHostTestSuite(t, "Bob")
+	eve := doctorFord.getHostTestSuite(t, "Eve")
+
+	ctxh := testingconfig.CreateAccountContext(t, eve.host.config)
+
+	// Get PublicKey and PrivateKey
+	publicKey, privateKey := GetSigningKeyPair(t, eve.host.idService, eve.id, ctxh)
+
+	// Revoke Key
+	key, err := utils.SliceToByte32(publicKey)
+	assert.NoError(t, err)
+	RevokeKey(t, eve.host.idService, key, eve.id, ctxh)
+
+	// Eve creates document with Bob and signs with Revoked key
+	collaborators := [][]byte{bob.id[:]}
+	dm := createCDWithEmbeddedPO(t, collaborators, eve.id, publicKey, privateKey)
+
+	signatures, signatureErrors, err := eve.host.p2pClient.GetSignaturesForDocument(ctxh, dm)
+	assert.NoError(t, err)
+	assert.Error(t, signatureErrors[0], "Signature verification failed error")
+	assert.Equal(t, 0, len(signatures))
+
+	// Bob creates document with Eve whose key is revoked
+	keys, err := eve.host.idService.GetKeysByPurpose(eve.id, &(identity.KeyPurposeSigning.Value))
+	assert.NoError(t, err)
+
+	// Revoke Key
+	RevokeKey(t, eve.host.idService, keys[0], eve.id, ctxh)
+
+	res := createDocument(bob.httpExpect, bob.id.String(), typeInvoice, http.StatusOK, defaultInvoicePayload([]string{eve.id.String()}))
+	txID := getTransactionID(t, res)
+	waitTillStatus(t, bob.httpExpect, bob.id.String(), txID, "failed")
```

Comment:
One minor thing, could you check if the transaction API call inside this function returns a message thats related to signature failure?